lib/module.cc

```diff
@@ -33,8 +36,12 @@ uintptr_t module::findModule(const char* moduleName, pid_t processId, const char
             continue;
         }
 
-        // We only need the first field, which is the starting address
-        address = strtoul(line, NULL, 16);
+       sscanf(line, "%" KLF "x-%" KLF "x %31s %llx %x:%x %llu", &result.start,
+			&result.end, result.permissions, &result.offset, 
+            &result.dev_major, &result.dev_minor, &result.inode);
+        
+        result.pathname = strchr(line,'/');
+
         found = true;
         break;
     }
```

Comment:
I think this `result.pathname` pointer is broken -- `line` is overwritten in each iteration of the loop, can be reallocated, and is `free`'d at the end of the loop.

- You could `strdup` it, but you would need to be careful to always clean it up. (You could make Module into a class, and add a destructor?)
- You could change the struct field to `char pathname[PATH_MAX];` (or 4096, or some other arbitrarily large number) -- but there are still edge cases where you *can* have paths that are longer than this.

Reply:
I used `strdup` for the next version, the pointer will then be deleted by the Struct's destructor.